gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-scim/src/main/java/io/gravitee/am/gateway/handler/scim/service/impl/UserServiceImpl.java

```diff
@@ -385,10 +406,10 @@ public Single<User> patch(String userId, PatchOp patchOp, String idp, String bas
 
                     // check password
                     if (isInvalidUserPassword(userToPatch.getPassword(), UserMapper.convert(userToPatch))) {
-                        return Single.error(new InvalidValueException("Field [password] is invalid"));
+                        return Single.error(new InvalidValueException(FIELD_PASSWORD_IS_INVALID));
                     }
 
-                    return update(userId, userToPatch, idp, baseUrl, principal);
+                    return update(userId, userToPatch, idp, baseUrl, principal, null);
```

Comment:
Instead of null, you should pass the client. 
In the same way as the update method, you have to provide the client as parameter at line 398 and then update the UserEnpoint to provide the value using the routingContext. like you did here https://github.com/gravitee-io/gravitee-access-management/pull/2188/files#diff-64e06ae1c40a661a2f1d9502a04eef338de3414e06558f42d0a6378191f7e49aR134

Reply:
This is why I suggested to you that these SCIM changes should have been pulled out into a separate subtask - that should have become clear when I started having to change interface method signatures. I'm now making changes to code that itself was created from changes you suggested in this PR.